@here/harp-mapview/lib/geometry/VertexCache.ts

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright (C) 2020 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Vector3Like } from "@here/harp-geoutils";
+import { assert } from "@here/harp-utils";
+
+// Offsets for the fields stored in cache for each vertex.
+enum Field {
+    VertexId = 0,
+    OlderIdx, // Index in cache of the immediately older vertex.
+    NewerIdx, // Index in cache of the immediately newer vertex.
+    X,
+    Y,
+    Z,
+    Count
+}
+
+const Invalid: number = -1;
+
+/**
+ * Compact vertex LRU Cache for on the fly temporary mesh transformations.
+ * @internal
+ */
+export class VertexCache {
+    private m_cache: number[] = []; // Stores all fields for every cached vertex (see Field).
+    private m_vertexCount: number = 0;
+    private m_oldestIdx: number = 0;
+    private m_newestIdx: number = 0;
+
+    /**
+     * Creates a new cache with the specified maximum size.
+     * @param maxVertexCount The maximum number of vertices the cache will store.
+     */
+    constructor(public maxVertexCount: number) {
```

Comment:
Minor: Make maxVertexCount readonly(if cache does not need to be resized) or implement it with a property so that `m_cache` is also resized when `maxVertexCount` is changed. The array should be automatically resized when data is inserted out of bounds but that may be slightly slower. It will also not actually get smaller if the cache size is decreased.

Reply:
I actually didn't plan to support variable size, so yes I'll make it readonly.